qiskit_nature/algorithms/point_generators/mp2_point_generator.py

```diff
@@ -0,0 +1,216 @@
+        self.threshold = threshold
+
+        self._terms = self._compute_corrections()
+
+    @property
+    def num_spin_orbitals(self) -> int:
+        """The number of spin orbitals."""
+        return self._num_spin_orbitals
+
+    @num_spin_orbitals.setter
+    def num_spin_orbitals(self, num_spin_orbs: int) -> None:
+        """The number of spin orbitals.
+
+        Raises:
+            ValueError: For non-positive-definite input.
+        """
+        if num_spin_orbs <= 0:
+            raise ValueError("Number of spin orbitals must be positive definite.")
+        self._num_spin_orbitals = num_spin_orbs
+
+    @property
+    def electronic_energy(self) -> ElectronicEnergy:
+        """The ElectronicEnergy property."""
+        return self._electronic_energy
+
+    @electronic_energy.setter
+    def electronic_energy(self, elec: ElectronicEnergy) -> None:
+        """The ElectronicEnergy property.
+
+        Raises:
+            ValueError: If ElectronicEnergy is missing orbital energies.
+        """
+        orbital_energies = elec.orbital_energies
+        if orbital_energies is None:
+            raise ValueError("Orbital energies missing from ElectronicEnergy.")
+
+        self._electronic_energy = elec
+        self._orbital_energies = orbital_energies
+        self._integral_matrix = elec.get_electronic_integral(ElectronicBasis.MO, 2).get_matrix()
+        self._reference_energy = elec.reference_energy
+
+    @property
+    def excitations(self) -> List[Tuple[Tuple[int, ...], Tuple[int, ...]]]:
+        """The sequence of excitations."""
+        return self._excitations
+
+    @excitations.setter
+    def excitations(self, exs: List[Tuple[Tuple[int, ...], Tuple[int, ...]]]):
+        """The sequence of excitations."""
+        # TODO: Validate excitation list as is done in UCC?
+        self._excitations = exs
+
+    @property
+    def threshold(self) -> float:
+        """The energy threshold for MP2 computation."""
+        return self._threshold
+
+    @threshold.setter
+    def threshold(self, thr: float) -> None:
+        """The energy threshold for MP2 computation."""
+        if thr <= 0:
+            raise ValueError("The energy threshold must be positive definite.")
+        self._threshold = thr
+
+    @property
+    def initial_point(self) -> np.ndarray:
+        """The MP2 coefficients as an initial_point."""
+        return np.asarray([val[0] for val in self._terms.values()])
+
+    @property
+    def energy_deltas(self) -> np.ndarray:
+        """The MP2 energy correction deltas for each excitation."""
+        return np.asarray([val[1] for val in self._terms.values()])
+
+    @property
+    def energy_delta(self) -> float:
+        """The MP2 delta energy correction for the molecule."""
+        return sum(self.energy_deltas)
+
+    @property
+    def energy(self) -> float:
+        """The absolute MP2 energy for the molecule."""
+        return self._reference_energy + self.energy_delta
```

Comment:
I am not sure I follow with the need - its not energy which is extracted like the transformers do right which alters the value we will find and needs to be added back at the end, To me its more like the HF energy as in its a value that is computed along the way. Although we set the MP2 informed initial point, I don't think this ends up evaluating as MP2 energy from VQE in the same way the HF starting point results in HF energy.

Reply:
Ah yes you are right! :thinking: 
So this is more an extra piece of information rather than a partial energy which the user needs to take into account in the end, right?

Reply:
Yes, just information if you want it. So I would not say they "need to take this into account" any more than the HF ref energy that is available if its wanted.